network/peers/connections/filters.go

```diff
@@ -1,16 +1,64 @@
 package connections
 
 import (
+	"crypto"
+	"crypto/rsa"
+
 	"github.com/bloxapp/ssv/network/records"
+	"github.com/bloxapp/ssv/operator/storage"
+	"github.com/libp2p/go-libp2p/core/peer"
 	"github.com/pkg/errors"
+	"go.uber.org/zap"
 )
 
 // NetworkIDFilter determines whether we will connect to the given node by the network ID
 func NetworkIDFilter(networkID string) HandshakeFilter {
-	return func(ni *records.NodeInfo) (bool, error) {
-		if networkID != ni.NetworkID {
-			return false, errors.Errorf("networkID '%s' instead of '%s'", ni.NetworkID, networkID)
+	return func(sender peer.ID, sni *records.SignedNodeInfo) (bool, error) {
+		if networkID != sni.NetworkID {
+			return false, errors.Errorf("networkID '%s' instead of '%s'", sni.NetworkID, networkID)
+		}
+		return true, nil
+	}
+}
+
+func SenderRecipientIPsCheckFilter(me peer.ID) HandshakeFilter {
+	return func(sender peer.ID, sni *records.SignedNodeInfo) (bool, error) {
+		if sni.HandshakeData.RecipientPeerID != me {
+			return false, errors.Errorf("recepient peer ID '%s' instead of '%s'", sni.HandshakeData.RecipientPeerID, me)
 		}
+
+		if sni.HandshakeData.SenderPeerID != sender {
+			return false, errors.Errorf("sender peer ID '%s' instead of '%s'", sni.HandshakeData.SenderPeerID, sender)
+		}
+
+		return true, nil
+	}
+}
+
+func SignatureCheckFilter(logger *zap.Logger, nodeStorage storage.Storage) HandshakeFilter {
+	privateKey, found, err := nodeStorage.GetPrivateKey()
```

Comment:
you need to verify the data with the Pubkey of the sender, not with our own pubkey. currently this code will always fail.

Reply:
Oh yeah, developer's mistake. 
That's why we always need tests ;)

Reply:
Again this is not solved, you are verifying the sender public key with our own public key. this shouldn't work and not what we're trying to do.